CareKit/CareKit/iOS/Task/View Controllers/OCKSurveyTaskViewController.swift

```diff
@@ -133,65 +133,81 @@ open class OCKSurveyTaskViewController: OCKTaskViewController<OCKSurveyTaskViewS
         didCompleteEvent isComplete: Bool,
         at indexPath: IndexPath,
         sender: Any?) {
-
-        guard isComplete else {
-
-            if let event = controller.eventFor(indexPath: indexPath),
-
-               let delegate = surveyDelegate,
-
-               delegate.surveyTask(
-                    viewController: self,
-                    shouldAllowDeletingOutcomeForEvent: event) == false {
-
-                return
-            }
-
-            let cancelAction = UIAlertAction(
-                title: "Cancel",
-                style: .cancel,
-                handler: nil
-            )
-
-            let confirmAction = UIAlertAction(
-                title: "Delete", style: .destructive) { _ in
+    
+            guard isComplete else {
+                
+                if let event = controller.eventFor(indexPath: indexPath),
+                   
+                    let delegate = surveyDelegate,
+                   
+                    delegate.surveyTask(
+                        viewController: self,
+                        shouldAllowDeletingOutcomeForEvent: event) == false {
+                    
+                    return
+                }
+                
+                let cancelAction = UIAlertAction(
+                    title: "Cancel",
+                    style: .cancel,
+                    handler: nil
+                )
+                
+                let confirmAction = UIAlertAction(
+                    title: "Delete", style: .destructive) { _ in
+                        
+                        super.taskView(
+                            taskView,
+                            didCompleteEvent: isComplete,
+                            at: indexPath,
+                            sender: sender
+                        )
+                    }
                 
-                super.taskView(
-                    taskView,
-                    didCompleteEvent: isComplete,
-                    at: indexPath,
-                    sender: sender
+                let warningAlert = UIAlertController(
+                    title: "Delete",
+                    message: "Are you sure you want to delete your response?",
+                    preferredStyle: .actionSheet
                 )
+                
+                warningAlert.addAction(cancelAction)
+                warningAlert.addAction(confirmAction)
+                
+                /*
+                 TODO: Remove in the future. Explicitly setting the tint color here to support
+                 current developers that have a SwiftUI lifecycle app and wrap this view
+                 controller in a `UIViewControllerRepresentable` implementation...Tint color
+                 is not propagated...etc.
+                 */
+                warningAlert.view.tintColor = determineTintColor(from: view)
```

Comment:
> The hope is to move away from wrapping our UIKit views for SwiftUI in part because doing so causes some unintended side effects (card sizing issues, tint color issues, etc.). There are definitely still tools we'll need to add to the framework to make the process even easier, but I'm hoping we can slowly to identify those gaps and build those up together. What do you think of that direction? Should we encourage people down that path and remove patches like this?

I believe this a good idea for the future as I eventually would like to use CareKit and SwiftUI entirely. My concern now is that the SwiftUI cards/views are not close enough in parity with the UIKit views. For example the Contact cards, SurveyTaskView, and WeekHeader. In addition, there's still reliance on asking for permissions, ResearchKit surveys, and opening other OS specific views such as the contact list which all seem to be UIKit based.

When I describe the reasons I use CareKit to build health care apps, I mention that CareKit allows for fast, themeable, attractive, offline-first, distributed based applications. Since most of the UIKit and SwiftUI views do different things, we lose some of the "fast" and "themeable" aspects because developers have to spend time building more custom SwiftUI views to handle what the UIKit cards currently cover and have colors that don't aline with the theme of an app.

In summary, I think the little bit of code to insure the views look the same across apps is a major upside to building current apps with CareKit. As the SwiftUI and UIKit views/cards come closer to parity, I think we can remove